migrations/2020-08-25-101655_lock_users/up.sql

```diff
@@ -0,0 +1,3 @@
+ALTER TABLE users
+    ADD COLUMN account_lock_reason VARCHAR DEFAULT NULL,
+    ADD COLUMN account_lock_until TIMESTAMP WITH TIME ZONE DEFAULT NULL;
```

Comment:
Is there a reason why you'd prefer not to include a timezone?

Personally I think not having TZs in the other timestamps is not a great idea either, since it's not clear in which TZ the timestamps were made. Changing all existing ones is a hassle, but we could at least start adding TZs in new ones.

Reply:
I haven't looked much into this, but my general impression was that it was a best practice to store everything in UTC and convert to local time zones elsewhere. For instance, when a new crate version is published, we don't know the user's local time zone and don't really have any need to track that. And while git does have the ability to encode a timezone offset, we never include an offset in the index. (For the git case in general, where people can commit in their local timezone and later consolidate/merge distributed changes, I think tracking the user's time zone does make sense.)

Having said that, including the time zone in this field is probably fine, especially with the assumption that this is rare enough that a mod would insert the row directly. In that case it makes sense that the mod adding the record can offload any timezone conversion to the database.

Reply:
I don't think the timezone is needed. The whole codebase represents times in UTC, which is a reasonable approach in my opinion. Postgres clients (e.g. `psql` or pgAdmin) treat times without a timezone as UTC as well, so this is not a convention we made up.

When designing this from scratch, I can see some merit in either approach, though overall using UTC everywhere and not bothering with timezones at all seems simpler. And we are not even designing this from scratch. We have a codebase with an existing convention. We use `NaiveDateTime` everywhere. We can discuss abandoning the convention and converting the codebase, but I don't see merit it just doing something else for new timestamps – this just makes things inconsistent and confusing.

Reply:
I hadn't seen Justin's comment when I wrote mine. It's true that this may be slightly different since it's entered manually in the DB, but I'd generally enter that as `NOW() + INTERVAL '1 MONTH'` or similar, so we don't really need timezones there either. And in the long run, I'd prefer to have functions like this in some kind of admin API, so we don't need to modify the production DB directly, which isn't really the best practice. :)

Reply:
Switched to `NaiveDateTime`.